src/main/scala/com/ibm/couchdb/api/builders/ViewQueryBuilder.scala

```diff
@@ -133,11 +131,37 @@ case class ViewQueryBuilder[K, V](client: Client,
     queryByIds[CouchDocs[K, V, D]](keys, includeDocs().params)
   }
 
-  private def queryWithoutIds[Q: R](ps: Map[String, String]): Task[Q] = {
-    query[Q](client, db, s"/$db/_design/$design/_view/$view", ps)
+  private def queryWithoutIds[Q: R](ps: Map[String, String]): Task[Q] = temporaryView match {
+    case Some(t) => query[CouchView, Q](client, db, url, t, ps)
+    case None => query[Q](client, db, url, ps)
   }
 
   private def queryByIds[Q: R](ids: Seq[K], ps: Map[String, String]): Task[Q] = {
-    queryByIds[K, Q](client, db, s"/$db/_design/$design/_view/$view", ids, ps)
+    if (ids.isEmpty)
+      Res.Error("not_found", "No IDs specified").toTask
+    else {
+      temporaryView match {
+        case Some(t) => query[DocViewWithKeys[K], Q](client, db, url, DocViewWithKeys(keys = ids, t), ps)
```

Comment:
I like to qualify the request types with `Req.`, like `Req.DocViewWithKeys` -- this makes it clearer that it's a part of a request.